commands/http/handler.go

```diff
@@ -16,10 +18,17 @@ import (
 
 var log = u.Logger("commands/http")
 
+// the internal handler for the API
+type internalHandler struct {
+	ctx  cmds.Context
+	root *cmds.Command
+}
+
+// The Handler struct is funny because we want to wrap our internal handler
+// with CORS while keeping our fields.
```

Comment:
not sure this is the best way, maybe @cryptix can review this too, he's been poking at the gateway